parsers/GT.py

```diff
@@ -32,21 +32,28 @@ def fetch_consumption(
     """Fetch a list of hourly consumption data, in MW, for the day of the
     requested date-time.
     """
-    date_time = arrow.get(target_datetime).to(TIMEZONE).floor("hour")
+
+    target_datetime = (
+        datetime.now(TIMEZONE)
+        if target_datetime is None
+        else target_datetime.astimezone(TIMEZONE)
+    )
+
+    api_data = _get_api_data(session, URL, target_datetime, target="consumption")
+
+    day_datetime = datetime.combine(
+        target_datetime, time(), tzinfo=TIMEZONE
+    )  # truncate to day
     results = [
         {
             "consumption": row["DEM SNI"],
-            "datetime": date_time.replace(hour=hour).datetime,
+            "datetime": day_datetime + timedelta(hours=hour),
```

Comment:
Any reason you went with timedelta instead of replace?

Reply:
I mostly just thought it expressed a bit more intuitively the logic.

The previous code was flooring the date_time to "hour" and then replacing that hour with those returned by the API, which confused me slightly. I thought it would be slightly more intuitive to start with today's day and iteratively step through hour by hour.